ivy/functional/ivy/experimental/elementwise.py

```diff
@@ -1217,3 +1217,59 @@ def modf(
     (ivy.array([0.5, 0.7, 0.9]), ivy.array([1, 2, 3]))
     """
     return ivy.current_backend(x).modf(x, out=out)
+
+
+@handle_exceptions
+@handle_nestable
+@handle_array_like_without_promotion
+@to_native_arrays_and_back
+def logcumsumexp(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    axis: Optional[int] = None,
+    dtype: Optional[str] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Return the logarithm of the cumulative summation of the exponentiation of the
+    elements along a given axis.
+
+    Parameters
+    ----------
+    x
+        Input array.
+    axis
+        The dimension to do the operation along. -1 means the last dimension.
+        The default (None) is to compute the cumsum over the flattened array.
+    dtype
+        The data type of the output array. This can be float32 or float64.
+        If specified, the input tensor is casted
+        to dtype before the operation is performed.
+        This can help prevent data type overflows.
+        The default value is None.
+    out
+        Optional array for storing the result to.
+
+    Returns
+    -------
+    ret
+        A tensor consisting of the result of the logcumsumexp operator
+
+    Examples
+    --------
+    >>> x = ivy.arrange(12, dtype='float64')
+    >>> data = ivy.reshape(x, (3, 4))
+    >>> ivy.logcumsumexp(data)
+    [ 0.         1.3132617  2.4076061  3.4401898  4.4519143  5.4561934
+    6.4577627  7.4583397  8.458551   9.45863   10.458658  11.458669 ]
+    >>> ivy.logcumsumexp(data, axis=0)
+    [[ 0.        1.        2.        3.     ]
+    [ 4.01815   5.01815   6.01815   7.01815 ]
+    [ 8.018479  9.018479 10.018479 11.018479]]
+    >>> ivy.logcumsumexp(data, axis=1)
+    # [[ 0.         1.3132617  2.4076061  3.4401898]
+    #  [ 4.         5.3132615  6.407606   7.44019  ]
+    #  [ 8.         9.313262  10.407606  11.440189 ]]
+    """
+    return ivy.current_backend(x).logcumsumexp(x, axis=axis, dtype=dtype, out=out)
```

Comment:
Supported frameworks don't have `logcumsumexp` fn so this wouldn't work. E.g. if you search `def logcumsumexp`, you'd find it doesn't exist in any of the backends